### PR TITLE
[vectorized] [block] add new method get_data_type to avoid unnecessary copy  by the method get_data_type

### PR DESCRIPTION
**Pull request by zuochunwei**
## Proposed changes
1. add new method get_data_type for class Block to reduce the copy cost of get_data_types
2. change function erase_useless_column()'s loop control variable i to type int for safety reason

Close related #issue (replace it with issue number if it exists).

Describe the overview of changes, and introduce why we need it.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ x] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by wangbo — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
perfectly

File: be/src/vec/core/block.h [hunk 1]
```diff
@@ -129,6 +129,7 @@ class Block {
 
     Names get_names() const;
     DataTypes get_data_types() const;
+    DataTypePtr get_data_type(size_t index) const { return index < data.size() ? data[index].type : nullptr; }
```
Comment (morningman) [inline be/src/vec/core/block.h:132]: Better to use `CHECK(index < data.size())`?

Comment (zuochunwei) [reply be/src/vec/core/block.h:132]: OK, i will do what you said, removing condition check will execute quicker

**Review by morningman — APPROVED** [review]

**Pull request merged**